config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -424,6 +431,26 @@ private String generateBundleFileName(boolean isDeleteBundle, String bundleName)
         return bundleName + filenameSuffix;
     }
 
+    public void addPrivateKeyContexts(Bundle bundle, ProjectInfo projectInfo, BundleArtifacts bundleArtifacts,
+                                      Document document) {
+        if (!bundle.getPrivateKeys().isEmpty()) {
+            bundle.getPrivateKeys().forEach((alias, privateKey) -> {
+                if (privateKey.getPrivateKeyFile() == null) {
+                    privateKey.setPrivateKeyFile(bundle.getPrivateKeyFiles().get(privateKey.getAlias()));
+                }
+                if (privateKey.getPrivateKeyFile() != null) {
+                    Element element = privateKeyImportContextBuilder.build(privateKey, document);
+                    String filename = generatePrivateKeyFileName(privateKey, projectInfo);
+                    bundleArtifacts.addPrivateKeyContext(element, filename);
+                }
+            });
+        }
+    }
+
+    private String generatePrivateKeyFileName(PrivateKey privateKey, ProjectInfo projectInfo) {
+        return projectInfo.getName() + "-" + privateKey.getAlias() + "-" + projectInfo.getVersion() + ".privatekey.xml";
```

Comment:
do we have to check for blank version, for other bundle names version is checked for blank

Reply:
for environment bundle we append config name, is it required for private key bundle ?

Reply:
@graju256 your thoughts on this naming?

Reply:
YES, I agree with Srini. Handle blank version and version with config names as we did it for bundles.

Reply:
@graju256 @uppoju Fixed it!